bottlecap/src/telemetry/listener.rs

```diff
@@ -15,6 +12,25 @@ pub struct HttpRequestParser {
     body: String,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum TcpError {
+    Close(String),
+    Read(String),
+    Write(String),
+    Parse(String),
+}
+
+impl ToString for TcpError {
```

Comment:
I don't think the name is adequate in this context, because `Parse` doesn't go with the TCP context, would it make sense to rename it? WDYT?

Reply:
I guess I interpreted it as we can't parse the message on the TCP socket, but wdyt?

Reply:
Not 100% convinced, but I understand the interpretation, we can leave it as is for now